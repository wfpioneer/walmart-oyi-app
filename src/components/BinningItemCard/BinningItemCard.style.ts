import { StyleSheet } from 'react-native';
import COLOR from '../../themes/Color';

export default StyleSheet.create({
  container: {
    alignItems: 'center',
    backgroundColor: COLOR.WHITE,
    justifyContent: 'space-between',
    flexDirection: 'row',
    flex: 1
  },
  firstLineContainer: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    marginBottom: 5
  },
  infoContainer: {
    flexDirection: 'column',
    margin: 10,
    flex: 4
  },
  lastLocationContainer: {
    backgroundColor: COLOR.YELLOW
  },
  icon: {
    paddingBottom: 5,
    paddingTop: 5,
    paddingLeft: 5,
    margin: 10,
    flex: 1
  },
  emptyPalletMsg: {
    color: COLOR.RED_500
  }
});
