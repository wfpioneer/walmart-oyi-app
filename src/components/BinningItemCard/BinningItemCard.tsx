/* eslint-disable react/require-default-props */
import React from 'react';
import {
  Text,
  TouchableOpacity,
  View
} from 'react-native';
import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import { strings } from '../../locales';
import styles from './BinningItemCard.style';
import COLOR from '../../themes/Color';

interface Props {
  palletId: number;
  itemDesc: string;
  lastLocation?: string;
  canDelete?: boolean;
  onClick?(): void;
  onDelete?(): void;
}

export const BinningItemCard = (props: Props): JSX.Element => {
  const {
    palletId, itemDesc, lastLocation, onClick, onDelete, canDelete
  } = props;
  return (
    <TouchableOpacity onPress={() => onClick && onClick()}>
      <View style={styles.container}>
        <View style={styles.infoContainer}>
          <View style={styles.firstLineContainer}>
            <View>
              <Text>{`${strings('PALLET.PALLET_ID')}: ${palletId}`}</Text>
            </View>
            {lastLocation
            && (
            <View style={styles.lastLocationContainer}>
              <Text>{`${strings('BINNING.LAST_LOC')}: ${lastLocation}`}</Text>
            </View>
            )}
          </View>
          <View>
            {itemDesc
              ? <Text>{`${strings('BINNING.FIRST_ITEM')}: ${itemDesc}`}</Text>
              : <Text style={styles.emptyPalletMsg}>{`${strings('BINNING.EMPTY_PALLET')}`}</Text>}
          </View>
        </View>
        {canDelete
        && (
        <View>
          <TouchableOpacity style={styles.icon} onPress={() => onDelete && onDelete()}>
            <View>
              <Icon name="trash-can" size={40} color={COLOR.TRACKER_GREY} />
            </View>
          </TouchableOpacity>
        </View>
        )}
      </View>
    </TouchableOpacity>
  );
};

export default BinningItemCard;
