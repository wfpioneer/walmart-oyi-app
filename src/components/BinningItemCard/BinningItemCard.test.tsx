import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import BinningItemCard from './BinningItemCard';

describe('Binning Item card render tests', () => {
  it('renders the binning item card', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BinningItemCard
        palletId={1}
        itemDesc="itemDesc"
        lastLocation="A1-1"
        canDelete={true}
        onClick={jest.fn}
        onDelete={jest.fn}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('renders the binning item card without last loc', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BinningItemCard
        palletId={1}
        itemDesc="itemDesc"
        canDelete={true}
        onClick={jest.fn}
        onDelete={jest.fn}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
  it('renders the binning item card in read only mode', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(
      <BinningItemCard
        palletId={1}
        itemDesc="itemDesc"
        lastLocation="A1-1"
        onClick={jest.fn}
        onDelete={jest.fn}
      />
    );
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
