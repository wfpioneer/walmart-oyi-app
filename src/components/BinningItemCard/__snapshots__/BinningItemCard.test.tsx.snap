// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Binning Item card render tests renders the binning item card 1`] = `
<ForwardRef
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffffff",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "column",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 5,
          }
        }
      >
        <View>
          <Text>
            [missing "en.PALLET.PALLET_ID" translation]: 1
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFE0",
            }
          }
        >
          <Text>
            [missing "en.BINNING.LAST_LOC" translation]: A1-1
          </Text>
        </View>
      </View>
      <View>
        <Text>
          [missing "en.BINNING.FIRST_ITEM" translation]: itemDesc
        </Text>
      </View>
    </View>
    <View>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "flex": 1,
            "margin": 10,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingTop": 5,
          }
        }
      >
        <View>
          <mockConstructor
            color="#545F7A"
            name="trash-can"
            size={40}
          />
        </View>
      </ForwardRef>
    </View>
  </View>
</ForwardRef>
`;

exports[`Binning Item card render tests renders the binning item card for empty pallet 1`] = `
<ForwardRef
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffffff",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "column",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 5,
          }
        }
      >
        <View>
          <Text>
            [missing "en.PALLET.PALLET_ID" translation]: 1
          </Text>
        </View>
      </View>
      <View>
        <Text
          style={
            Object {
              "color": "#f44336",
            }
          }
        >
          [missing "en.BINNING.EMPTY_PALLET" translation]
        </Text>
      </View>
    </View>
    <View>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "flex": 1,
            "margin": 10,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingTop": 5,
          }
        }
      >
        <View>
          <mockConstructor
            color="#545F7A"
            name="trash-can"
            size={40}
          />
        </View>
      </ForwardRef>
    </View>
  </View>
</ForwardRef>
`;

exports[`Binning Item card render tests renders the binning item card in read only mode 1`] = `
<ForwardRef
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffffff",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "column",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 5,
          }
        }
      >
        <View>
          <Text>
            [missing "en.PALLET.PALLET_ID" translation]: 1
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#FFFFE0",
            }
          }
        >
          <Text>
            [missing "en.BINNING.LAST_LOC" translation]: A1-1
          </Text>
        </View>
      </View>
      <View>
        <Text>
          [missing "en.BINNING.FIRST_ITEM" translation]: itemDesc
        </Text>
      </View>
    </View>
  </View>
</ForwardRef>
`;

exports[`Binning Item card render tests renders the binning item card without last loc 1`] = `
<ForwardRef
  onPress={[Function]}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffffff",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "column",
          "margin": 10,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 5,
          }
        }
      >
        <View>
          <Text>
            [missing "en.PALLET.PALLET_ID" translation]: 1
          </Text>
        </View>
      </View>
      <View>
        <Text>
          [missing "en.BINNING.FIRST_ITEM" translation]: itemDesc
        </Text>
      </View>
    </View>
    <View>
      <ForwardRef
        onPress={[Function]}
        style={
          Object {
            "flex": 1,
            "margin": 10,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingTop": 5,
          }
        }
      >
        <View>
          <mockConstructor
            color="#545F7A"
            name="trash-can"
            size={40}
          />
        </View>
      </ForwardRef>
    </View>
  </View>
</ForwardRef>
`;
